pallets/sto/src/lib.rs

```diff
@@ -282,7 +282,7 @@ decl_module! {
             let offering_amount: T::Balance = tiers
                 .iter()
                 .map(|t| t.total)
-                .fold(0.into(), |total, x| total + x);
```

Comment:
This shouldn't be checked instead with an error instead?

Reply:
That was my first thought, and then I find the following case:
1. Let's  asume balance = u8, so max = 255.
2. User creates the folling `PriceTiers`: 
```
[
  { total= 10, price = 1},
  { total = 20, price = 2},
  { total = MAX = 255, price = 3 }
]
```
  
In the current implementation, the last case is still valid an it is like "rest of all at 3" and it locks 255 at the fundraiser.
It's a weird case, but I would not moderate as an error.

Any thoughts?

Reply:
I think we should error here.
This case for example, the latter tiers are essentially useless and IMO we should reject this.
In general I do agree with @fmiguelgarcia that overflowing is fine because both balance and the supply of an asset are u128 so if you have a tier with a u128::MAX that will saturate the assets supply before overflowing.
```
[
  { total= MAX, price = 1},
  { total = 10, price = 2},
  { total = 20 = 255, price = 3 }
]
```

Reply:
Let's throw an error here.

Reply:
Thanks team!, I've added an error when total overflow and its UT.